src/NearestNeighborsModule.jl

```diff
@@ -0,0 +1,65 @@
+module NearestNeighborsModule
+
+using KernelFunctions, LinearAlgebra, SparseArrays, AbstractGPs
+
+function make_B(pts::AbstractVector{T}, k::Int, kern::Kernel) where {T}
+	n = length(pts)
+	js = Int[]
+	is = Int[]
+	vals = T[]
+	for i in 1:n
+		if i == 1
+			ns = T[]
+		else
+			ns = pts[max(1, i-k):i-1]
+		end
+		row = kernelmatrix(kern,ns) \ kern.(ns, pts[i])
+		start_ix = max(i-k, 1)
+		col_ixs = start_ix:(start_ix + length(row) - 1)
+		append!(js, col_ixs)
+		append!(is, fill(i, length(col_ixs)))
+		append!(vals, row)
+	end
+	sparse(is, js, vals, n, n)
+end
+
+function make_F(pts::AbstractVector{T}, k::Int, kern::Kernel) where {T}
+	n = length(pts)
+	vals = T[]
+	for i in 1:n
+		prior = kern(pts[i], pts[i])
+		if i == 1
+			push!(vals, prior)
+		else
+			ns = pts[max(1, i-k):i-1]
+			ki = kern.(ns, pts[i])
+			push!(vals, prior - dot(ki, kernelmatrix(kern, ns) \ ki))
+		end
+	end
+	Diagonal(vals)
+end
+
+struct NearestNeighbors
```

Comment:
Could you please add a docstring to explain what this approximation does, what the value of `k` determines, and to provide a reference / references?